packages/editor/src/components/provider/index.js

```diff
@@ -111,45 +147,79 @@ function useBlockEditorProps( post, template, mode ) {
 		useEntityBlockEditor( 'postType', template?.type, {
 			id: template?.id,
 		} );
-	const blocks = useMemo( () => {
-		if ( post.type === 'wp_navigation' ) {
-			return [
-				createBlock( 'core/navigation', {
-					ref: post.id,
-					// As the parent editor is locked with `templateLock`, the template locking
-					// must be explicitly "unset" on the block itself to allow the user to modify
-					// the block's content.
-					templateLock: false,
-				} ),
-			];
+	const postContentBlock =
+		extractPostContentBlockFromTemplateBlocks( templateBlocks );
+	const postContentLayout = usePostEditorLayout( postContentBlock );
```

Comment:
Just thinking, what about other style-based block attributes that could be set within a template, e.g. typography, min height, background color, and eventually background image support? Is it worth considering grabbing all the other attributes set on the `postContentBlock` in addition to figuring out the right layout attributes to use?